docker.js

```diff
@@ -125,49 +84,36 @@ function buildDockerCommand({
   return dockerArguments.join(" ");
 }
 
-function buildEnvironmentVariableArguments(environmentVariableNames) {
+function buildEnvironmentVariableArguments(environmentVariables) {
   if (
-    !environmentVariableNames
-    || !_.isArray(environmentVariableNames)
-    || _.some(environmentVariableNames, (variableName) => !_.isString(variableName))
+    !environmentVariables
+    || !_.isObject(environmentVariables)
+    || _.isArray(environmentVariables)
   ) {
-    throw new Error("Environment Variable Names parameter must be an array of strings.");
+    throw new Error("environmentVariables parameter must be an object.");
   }
 
-  return environmentVariableNames
-    .map((variableName) => ["-e", variableName])
+  return Object.entries(environmentVariables)
+    .map(([name, value]) => ["-e", `${name}=${value}`])
```

Comment:
We can pass environment variable values to the `exec` function, otherwise the environment variable value is directly visible in the shell command, which defeats the whole idea of using environment variables for additional security. We can just specify the environment variable names which we want to pass down into the container.

Reply:
Right,  good one.